### DISCUSSION TITLE
Different error meanings..

### DISCUSSION THREAD

**Discussion by bcookatpcsd** [DNSCrypt/dnscrypt-proxy#2112] [Q&A]
Looking for what the difference between PARSE_ERROR, NETWORK_ERROR, and SERVFAIL are.. 

Assuming nothing on my end.. 

```
assess-va.learnosity.com A NETWORK_ERROR 7ms provider-dnscrypt
events-va.learnosity.com A NETWORK_ERROR 7ms provider-doh
atopicderm.org A SERVFAIL 33ms provider-dnscrypt
e2c46.gcp.gvt2.com A PARSE_ERROR 4246ms -
fb.blooket.com A PARSE_ERROR 4193ms -
capi.grammarly.com A PARSE_ERROR 4011ms -
```

(provider is changed purposefully - but they offer dot, doh, dnscrypt, etc)
Timeout is 5000

Just wondering what is happening at 7ms that "NETWORK_ERROR" is the answer.. and 33ms "SERVFAIL" but 4000+ is PARSE_ERROR.. 

I have dnsdist in front of a dnscrypt-proxy .. 

![image](https://user-images.githubusercontent.com/55087301/167474240-ec5eefb5-17c5-44e0-9425-ebcd0047ab25.png)

and had a bunch of errors today..

**Comment by jedisct1** [comment]
`SERVFAIL` means that the configuration of the `atopicderm.org` domain is invalid. It's a broken DNSSEC configuration as usual. Google, Cloudflare, or any validating resolver is returns `SERVFAIL` in that case.

`NETWORK_ERROR` is a network error that is not a timeout. The certificate is invalid, the connection was closed by the server, the server is overloaded, you triggered some rate limit, etc.

`PARSE_ERROR` means that the received response is not a valid DNS response. When using DoH, it can be that the server returned some HTML error page instead of DNS.

**Reply by bcookatpcsd** [reply]
Thank you sir. 

Working with a different recursive provider.. trying to understand how what their doing appears inside dnscrypt-proxy. 

Greatly appreciated as always.
